spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/MongoTemplate.java

```diff
@@ -1088,7 +1089,7 @@ public <T> T findAndModify(Query query, UpdateDefinition update, FindAndModifyOp
 			operations.forType(entityClass).getCollation(query).ifPresent(optionsToUse::collation);
 		}
 
-		return doFindAndModify(createDelegate(query), collectionName, query.getQueryObject(), query.getFieldsObject(),
+		return doFindAndModify(createDelegate(query), collectionName, query, query.getFieldsObject(),
```

Comment:
In case this change is allowed, should I remove `query.getFieldsObject()`, since `query` is already being passed?

Reply:
I've changed the signature to remove redundant arguments to `doFindAndModify`. If the signature changes are to be kept to a minimum, I'll remove this commit.